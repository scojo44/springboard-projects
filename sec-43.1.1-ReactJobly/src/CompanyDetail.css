.CompanyDetail {
  display: flex;
  list-style-type: none;
}
.CompanyDetail .info {
  background-color: rgba(0,0,0,30%);
  width: 15em;
}
.CompanyDetail .jobs {
  margin: 1rem;
}
.CompanyDetail h3 {
  text-align: center;
  margin: 0;
  padding: .5rem;
}
.CompanyDetail p {
  margin: 0;
  padding: .5rem;
}
.CompanyDetail img {
  display: block;
  margin: auto;
  min-width: 90%;
  min-height: 8rem;
}
