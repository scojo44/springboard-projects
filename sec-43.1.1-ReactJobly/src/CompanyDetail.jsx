import React, {useEffect, useState} from 'react'
import {useParams} from 'react-router-dom'
import JoblyApi from './api'
import Alert from './Alert'
import './CompanyDetail.css'

export default function CompanyDetail() {
  const {handle} = useParams();
  const [company, setCompany] = useState([]);
  const [error, setError] = useState('');

  useEffect(() => {
    async function fetchCompany() {
      try {
        const firm = await JoblyApi.getCompany(handle);
        setCompany(() => firm);
      }
      catch(e) {
        setError(e);
      }
    }

    fetchCompany();
  }, []);


  const {name, description, logoURL} = company;

  return (
    <section className="CompanyDetail">
      <div className="info">
        <h3>{name}</h3>
        <img src={logoURL} />
        <p>{description}</p>
      </div>
      <ul className="jobs">
        {error && <Alert type="error" messages={[`Error loading companies: ${error}`]} />}
      </ul>
    </section>
  )
}
