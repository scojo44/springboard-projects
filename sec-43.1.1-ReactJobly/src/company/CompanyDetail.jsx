import React from 'react'
import {useParams} from 'react-router-dom'
import JoblyApi from '../api'
import useJoblyAPI from '../hooks/useJoblyAPI'
import LoadingIndicator from '../widgets/LoadingIndicator'
import Alert from '../widgets/Alert'
import JobCard from '../job/JobCard'
import './CompanyDetail.css'

export default function CompanyDetail() {
  const {data: company, error, isLoading} = useJoblyAPI(JoblyApi.getCompany, useParams().handle);
  const {name, description, numEmployees, logoURL, jobs} = company;

  if(isLoading) return <LoadingIndicator />;

  return (
    <section className="CompanyDetail">
      <div className="info">
        <h3>{name}</h3>
        <img src={logoURL} alt={name + ' Logo'} />
        <p>{description}</p>
        <p>Employees: <span className="detail-value">{numEmployees}</span></p>
      </div>
      {error
        ? <div><Alert type="error" messages={[`Error loading companies: ${error}`]} /></div>
        : <ul className="jobs">
            {jobs.map(j => <JobCard job={j} key={j.id} />)}
          </ul>
      }
    </section>
  );
}
