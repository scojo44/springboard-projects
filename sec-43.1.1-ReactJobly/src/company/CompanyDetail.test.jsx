import {getByAltText, render, waitForElementToBeRemoved} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import JoblyApi from '../api'
import CompanyDetail from './CompanyDetail'

const TEST_COMPANY = {
  handle: 'test-corp',
  name: 'Test Corp.',
  description: 'Test Corp is the leading.... yada yada yada',
  numEmployees: 5,
  logoURL: '/test-logo.png',
  jobs: []
}

// Mock the API hook
JoblyApi.getCompany = vi.fn(() => TEST_COMPANY);

describe('CompanyDetail Tests', () => {
  it('Renders without crashing', async () => {
    const {getByText} = render(
      <MemoryRouter>
        <CompanyDetail company={TEST_COMPANY} />
      </MemoryRouter>
    );

    await waitForElementToBeRemoved(getByText('Loading...'));
  });

  it('Matches snapshot', async () => {
    const {asFragment, getByText} = render(
      <MemoryRouter>
        <CompanyDetail company={TEST_COMPANY} />
      </MemoryRouter>
    );

    await waitForElementToBeRemoved(getByText('Loading...'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('Shows a company', async () => {
    const {getByText, getByAltText} = render(
      <MemoryRouter>
        <CompanyDetail company={TEST_COMPANY} />
      </MemoryRouter>
    );

    await waitForElementToBeRemoved(getByText('Loading...'));
    expect(getByText('5')).toBeInTheDocument();
    expect(getByText(TEST_COMPANY.name)).toBeInTheDocument();
    expect(getByText(TEST_COMPANY.description)).toBeInTheDocument();
    expect(getByAltText(TEST_COMPANY.name + ' Logo')).toBeInTheDocument();
  });
});
