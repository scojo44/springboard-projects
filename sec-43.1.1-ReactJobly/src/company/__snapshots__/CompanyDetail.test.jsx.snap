// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyDetail Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="CompanyDetail"
  >
    <div
      class="info"
    >
      <h3>
        Test Corp.
      </h3>
      <img
        alt="Test Corp. Logo"
        src="/test-logo.png"
      />
      <p>
        Test Corp is the leading.... yada yada yada
      </p>
      <p>
        Employees: 
        <span
          class="detail-value"
        >
          5
        </span>
      </p>
    </div>
    <ul
      class="jobs"
    />
  </section>
</DocumentFragment>
`;
